src/main/VGMColl.cpp

```diff
@@ -196,7 +197,19 @@ bool VGMColl::MainDLSCreation(DLSFile &dls) {
       VGMInstr *vgminstr = set->aInstrs[i];
       size_t nRgns = vgminstr->aRgns.size();
       std::string name = wstring2string(vgminstr->name);
-      DLSInstr *newInstr = dls.AddInstr(vgminstr->bank, vgminstr->instrNum, name);
+      auto bank_no = vgminstr->bank & 0x7f;
+      /*
+      * The ulBank field follows this structure:
+      * F|00000000000000000|CC0|CC32
+      * where F = 0 if the instrument is melodic, 1 otherwise
+      * (length of each CC is 7 bits, obviously)
+      */
+      if (auto bs = ConversionOptions::the().GetBankSelectStyle(); bs == BankSelectStyle::GS) {
+        bank_no = bank_no << 7;
+      } else if (bs == BankSelectStyle::MMA) {
+        bank_no = ((bank_no / 128) << 7) | bank_no;
+      }
```

Comment:
```suggestion
      auto bank_no = vgminstr->bank;
      /*
      * The ulBank field follows this structure:
      * F|00000000000000000|CC0|0|CC32
      * where F = 0 if the instrument is melodic, 1 otherwise
      * (length of each CC is 7 bits, obviously)
      */
      if (auto bs = ConversionOptions::the().GetBankSelectStyle(); bs == BankSelectStyle::GS) {
        bank_no &= 0x7f;
        bank_no = bank_no << 8;
      } else if (bs == BankSelectStyle::MMA) {
        const uint8_t bank_msb = (bank_no >> 7) & 0x7f;
        const uint8_t bank_lsb = bank_no & 0x7f;
        bank_no = (bank_msb << 8) | bank_lsb;
      }
```

The incorrect description of `ulBank` confused me a lot. 😞

With this change, I believe `VGMInstr` will have the same internal MMA-style bank number as `VGMSeq`, and map it properly during DLS export.

Before the change, the bank number of `VGMInstr` used the same format as the bank number of DLS. However, upon investigation, it turned out that only a small bank number was actually being used, and only `AkaoSnesDrumKit` was intentionally trying to use bank MSB.

I will fix bank mapping in AkaoSnesDrumKit in another comment.

Reply:
I wasn't sure where best to comment, so I'll post a diff here to fix the bank number for AkaoSnesDrumKit.

```diff
 src/main/formats/AkaoSnesInstr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/formats/AkaoSnesInstr.cpp b/src/main/formats/AkaoSnesInstr.cpp
index a0d9a59..04ba393 100644
--- a/src/main/formats/AkaoSnesInstr.cpp
+++ b/src/main/formats/AkaoSnesInstr.cpp
@@ -154,7 +154,7 @@ AkaoSnesDrumKit::AkaoSnesDrumKit(VGMInstrSet *instrSet,
                                  uint16_t addrADSRTable,
                                  uint16_t addrDrumKitTable,
                                  const std::wstring &name) :
-  VGMInstr(instrSet, addrTuningTable, 0, ((programNum >> 6) & 0x7F00) | ((programNum >> 7) & 0x7F), programNum & 0xFF, name), version(ver),
+  VGMInstr(instrSet, addrTuningTable, 0, programNum >> 7, programNum & 0x7F, name), version(ver),
   spcDirAddr(spcDirAddr),
   addrTuningTable(addrTuningTable),
   addrADSRTable(addrADSRTable),
 src/main/formats/AkaoSnesInstr.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/formats/AkaoSnesInstr.h b/src/main/formats/AkaoSnesInstr.h
index 0b48c3c..8b9d259 100644
--- a/src/main/formats/AkaoSnesInstr.h
+++ b/src/main/formats/AkaoSnesInstr.h
@@ -11,7 +11,7 @@
 class AkaoSnesInstrSet:
     public VGMInstrSet {
  public:
-  static const uint32_t DRUMKIT_PROGRAM = (0x7F << 14);
+  static const uint32_t DRUMKIT_PROGRAM = (0x7F << 7);
 
   AkaoSnesInstrSet(RawFile *file,
                    AkaoSnesVersion ver,
```

Before the change, bank MSB=127 was used, but after the change, bank LSB=127 will be used for the MMA style. It might be possible to abstract "bank for drums" so that it always prefers bank MSB=127, but for now we simply focused on fixing the bank number mismatch.

Reply:
Thank you for the review, will add this.